### PR TITLE
[flang][OpenMP] Handle unstructured CF in compound loop constructs

### PR DESCRIPTION
**Pull request by ergawy**
Fixes a bug in handling unstructured control-flow in compound loop constructs. The fix makes sure that unstructured CF does not get lowered until we reach the last item of the compound construct. This way, we avoid moving block of unstructured loops in-between the middle items of the construct and messing (i.e. adding operations) to these block while doing so.

### CODE REVIEW

**Review by skatrak — APPROVED** [review]
Thank you Kareem, this LGTM. Give it a day or wait for a second review before merging, in case I'm missing anything here.

**Review by tblah — APPROVED** [review]
LGTM too

**Pull request merged**
